config/examples/dot_product.yaml

```diff
@@ -0,0 +1,60 @@
+---
+type: config
+log_level: DEBUG
+cpu_affinity: [0,1,2,3]
+num_local_freq: 1
+samples_per_data_set: 4608
+
+# Constants
+sizeof_float: 4
+
+# Create pool of metadata for each buffer
+main_pool:
+  kotekan_metadata_pool: chimeMetadata
+  num_metadata_objects: 30
+
+# Construct input buffers
+input_buffers:
+  # Both input buffers will share the same 
+  # no. of frames and frame size
+  num_frames: 10
+  frame_size: 128 * sizeof_float
+  input_a_buffer:
+    kotekan_buffer: standard
+    metadata_pool: main_pool
+  input_b_buffer:
+    kotekan_buffer: standard
+    metadata_pool: main_pool
+
+# Construct output buffer
+output_buffer:
+  num_frames: 10
+  frame_size: 128 * sizeof_float
+  kotekan_buffer: standard
+  metadata_pool: main_pool
```

Comment:
I believe you can actually set the metadata pool to `none`, which makes this example simpler - and avoids CHIME specific code.

Reply:
When I set `metadata_pool: none` I get the following error:
```
Buffer: input_a_buffer Registering producer: /data_gen/data_gen_a
Buffer: input_b_buffer Registering producer: /data_gen/data_gen_b
Registering consumer /dot_product for buffer input_a_buffer
Registering consumer /dot_product for buffer input_b_buffer
Buffer: output_buffer Registering producer: /dot_product
Registering consumer /screen_dump for buffer output_buffer
restServer: Adding GET endpoint: /buffers
restServer: Adding GET endpoint: /pipeline_dot
Starting kotekan_stage: /data_gen/data_gen_a...
/data_gen/data_gen_a: Setting thread affinity
No metadata pool on input_a_buffer but metadata was needed by a producer
Starting kotekan_stage: /data_gen/data_gen_b...
/data_gen/data_gen_b: Setting thread affinity
Segmentation fault (core dumped)
```